spec/validators/ccd_address_validator_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+RSpec.describe CcdAddressValidator do
+  let(:valid_values) do
+    ['40 Station Road', '9209 Grange Road', '81 Highfield Road', '8 Broadway', '9691 The Avenue']
+  end
+  let(:invalid_values) do
+    ['40stationroadadresdefrtghyjuikonhgvfesadfgtreaqwdrf', '9209GrangeRoadadresdefrtghyjuikonhgvfesadfgtreaqwdrf', '81HighfieldRoadadresdefrtghyjuikonhgvfesadfgtreaqwdrf']
+  end
+
+  let(:model_class) do
+    Class.new do
+      include ActiveModel::Model
+      attr_accessor :example_address_street
+
+      def self.name
+        'MyModel'
+      end
+
+      validates :example_address_street, ccd_address: true
+    end
+  end
+
+  it 'is valid for all in the valid list' do
+    aggregate_failures 'validating all in valid list' do
+      valid_values.each do |value|
+        model = model_class.new(example_address_street: value)
+
+        model.valid?
+
+        expect(model.errors).to be_empty
+      end
+    end
+  end
+
+  it 'is valid for all downcased values in the valid list' do
+    aggregate_failures 'validating all in valid list' do
+      valid_values.each do |value|
+        model = model_class.new(example_address_street: value.downcase)
+
+        model.valid?
+
+        expect(model.errors).to be_empty, "expected #{value} to be valid but it wasn't"
+      end
+    end
+  end
+
+  it 'is not valid for nil' do
```

Comment:
The scenario name says it should not be valid but the scenario itself says its testing for no errors